ftms_filtering.qmd

```diff
@@ -130,101 +139,114 @@ filterPrecursorScan(al[[1]], 3919)
 #' Seems to be OK
 ```
 
-the resulting spectra object contains 1679 MS2, 1858 MS3, and 926 MS4.
-
-
+The number of spectra from different MS levels is shown below:
 
 ```{r}
-spectraData(ftms_levels)
+table(msLevel(ftms_msn_tree))
 ```
 
-
-
-
-This function creates MSntreeID column that gouped related scans into
+We next need to group all MSn spectra from one fragment tree together.
+The function below creates an *MSntreeID* column grouping related scans into
 fragmentation trees based on their parent-child relationships.
+
 ```{r}
 
-assign_msntree_id <- function(ftms_levels) {
-  scan_index <- scanIndex(ftms_levels)
-  prec_scan_num <- precScanNum(ftms_levels)
-  ms_level <- msLevel(ftms_levels)
-  data_origin <- dataOrigin(ftms_levels)
+assign_msntree_id <- function(x) {
+  ## Order x by MS level: ensures we first assign an ID to the MS2.
+  o <- order(msLevel(x))
```

Comment:
To explain @Ahlammentag : ordering the spectra first by MS level avoids the checks for MS level below in the for loop and allows to simplify the code considerably. So, for each tree, first MS2 spectra are evaluated, then all MS3 spectra and then all MS4 spectra. The code can thus assume that the ID of the precursor was already added.

At the end the results need to be re-ordered to match the original order.